src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -116,11 +116,19 @@ public Workspace Workspace
 
     /// <summary>
     /// A list of all the ids for all the projects contained by the solution.
+    /// Ordering determined by the order the projects were added to the solution.
     /// </summary>
     public IReadOnlyList<ProjectId> ProjectIds => this.SolutionState.ProjectIds;
 
+    /// <summary>
+    /// A list of all the project states contained by the solution.
+    /// Ordered by <see cref="ProjectState.Id"/>'s <see cref="ProjectId.Id"/> value.
+    /// </summary>
+    internal ImmutableArray<ProjectState> ProjectStates => this.SolutionState.ProjectStates;
```

Comment:
please call this SortedProjectStates.  It will help (imo) with readability everywhere and feeling safe around code that depends on that sortedness.  it will also make it abundantly clear if we break this what the contract is supposed to be.

Reply:
Done!